build_tools/cmake/iree_run_module_test.cmake

```diff
@@ -264,14 +260,14 @@ function(iree_benchmark_suite_module_test)
   cmake_parse_arguments(
     _RULE
     ""
-    "NAME;BENCHMARK_MODULE_SRC;MODEL;DRIVER;EXPECTED_OUTPUT;TIMEOUT"
+    "NAME;MODEL;DRIVER;EXPECTED_OUTPUT;TIMEOUT"
     "RUNNER_ARGS;LABELS;XFAIL_PLATFORMS;UNSUPPORTED_PLATFORMS"
     ${ARGN}
   )
 
-  # Benchmark suite needs to be stored at the location of
-  # `IREE_BENCHMARK_SUITE_DIR` or the test target is bypassed.
-  if(NOT DEFINED IREE_BENCHMARK_SUITE_DIR)
+  # Benchmark artifacts needs to be stored at the location of
+  # `IREE_E2E_TEST_ARTIFACTS_DIR` or the test target is bypassed.
+  if(NOT DEFINED IREE_E2E_TEST_ARTIFACTS_DIR)
```

Comment:
Seems like this should be gated on an enable flag. If someone enables these and then doesn't set the directory, that should be an error, no?

Reply:
The logic here is the same as `IREE_HOST_BINARY_ROOT`. it is not a flag, and if someone set the variable directly the cmake will follow the set condition, such as in 
https://github.com/iree-org/iree/blob/72e0d3c9468ef3659352efbbd8a2571785fd6a22/build_tools/cmake/test_benchmark_suites_host.sh#L33-L42

Reply:
I think the ideal way is to have a `IREE_ENABLE_E2E_MODEL_TESTS` to control it. And

- If `IREE_E2E_TEST_ARTIFACTS_DIR` is set, we use the pre-built artifacts from there
- If not, we build the artifacts by ourselves.

I'll address this in a follow-up change.